content/tutorials/portenta-h7/por-ard-flash/content.md

```diff
@@ -0,0 +1,256 @@
+# Using the Flash on Portenta H7 – MCU
+
+This tutorial shows you how to use the internal flash memory of the microcontroller of Portenta H7 and other MbedOS-based Arduino boards (Nano 33 BLEs, etc.). This is Part 1 of a series of projects on how to use the flash memories on the MbedOS-based Arduino boards.
+
+## (What) You Will Learn
+-   (How) to use the Flash In-Application Programming Interface 
+-   (How) to calculate the size of the flash memory 
+
+## Required Hardware and Software
+-   Portenta H7 board (<https://store.arduino.cc/portenta-h7>)
+-   USB C cable (either USB A to USB C or USB C to USB C)
+-   Arduino IDE 1.8.10+ or Arduino Pro IDE 0.0.4+ 
+
+
+# Configuring the Development Environment
+The H7 comes with 2 Mbytes of Flash memory with read-while-write support + 1 Mbyte of RAM. To support the use of non-erasable memory, a part of the flash memory can be programmed to behave as a block device, that can operate as a place to store information. 
+
+To access the internal flash of the MCU, you will have to use the Flash In-Application Programming Interface (FlashIAP). This API creates a block device on top of the space still available on the flash after flashing your sketch. Please**,** note that **the free space** available for the FlashIAP block device **depends on** the space occupied by **the complete sketch** on the flash and a few **block alignments calculations**. You can then use a raw access API to save and load data from the block device.
+
+https://os.mbed.com/docs/mbed-os/v6.4/apis/flash-iap.html
+
+Please, be aware of the **flash r/w limits** while using raw/direct access: flash memories have a limited amount of write cycles. Typical flash memories can perform about 10000 writes cycles to the same block before starting to "wear out" and begin to lose the ability to retain data. **You can actually render your board useless with improper use of this example and described APIs.**
```

Comment:
```suggestion
Please, be aware of the **flash r/w limits** while using raw/direct access: flash memories have a limited amount of write cycles. Typical flash memories can perform about 10000 writes cycles to the same block before starting to "wear out" and begin to lose the ability to retain data. **You can actually render your board useless with improper use of this example and described APIs.**
```
When saving and reading data from the flash memory, we need to be aware of the **flash r/w limits** while using raw/direct access: flash memories have a limited amount of write cycles. Typical flash memories can perform about 10000 writes cycles to the same block before starting to "wear out" and begin to lose the ability to retain data. **You can actually render your board useless with improper use of this example and described APIs.**